pkg/ccl/backupccl/restore_job.go

```diff
@@ -900,10 +904,20 @@ func createImportingDescriptors(
 		return nil, nil, nil, err
 	}
 
-	for _, desc := range tableDescs {
-		desc.Version++
-		desc.State = descpb.DescriptorState_OFFLINE
-		desc.OfflineReason = "restoring"
+	// Set the new descriptors' states to offline.
+	for _, desc := range mutableTables {
+		desc.SetOffline("restoring")
+	}
+	for _, desc := range typesToWrite {
+		desc.SetOffline("restoring")
+	}
+	for _, desc := range schemasToWrite {
+		desc.SetOffline("restoring")
+	}
+	if r.settings.Version.IsActive(ctx, clusterversion.VersionLeasedDatabaseDescriptors) {
```

Comment:
A small comment here about how we need this check since 20.1 nodes don't update databases from OFFLINE to PUBLIC would be useful.

Reply:
Done.